core/src/test/java/fj/control/db/TestDbState.java

```diff
@@ -0,0 +1,42 @@
+package fj.control.db;
+
+import fj.Unit;
+import fj.data.Option;
+import org.junit.Test;
+
+import java.sql.*;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+public class TestDbState {
+    @Test
+    public void testWriter() throws SQLException {
+        DbState writer = DbState.writer(DbState.driverManager("jdbc:h2:mem:"));
+
+        DB<Unit> setup = new DB<Unit>() {
+            @Override
+            public Unit run(Connection c) throws SQLException {
+                Statement s = c.createStatement();
+                assertThat(s.executeUpdate("CREATE TABLE TEST(ID INT PRIMARY KEY, NAME VARCHAR(255))"), is(0));
+                assertThat(s.executeUpdate("INSERT INTO TEST (ID, NAME) VALUES (10, 'FOO')"), is
+                        (1));
+                return Unit.unit();
+            }
+        };
+        DB<Option<Integer>> query = new DB<Option<Integer>>() {
+            @Override
+            public Option<Integer> run(Connection c) throws SQLException {
+                PreparedStatement ps = c.prepareStatement("SELECT ID FROM TEST WHERE NAME = ?");
+                ps.setString(1, "FOO");
+                ResultSet rs = ps.executeQuery();
+                if (rs.next()) {
+                    return Option.some(rs.getInt("ID"));
+                } else {
+                    return Option.none();
+                }
+            }
+        };
+        assertThat(writer.run(setup.bind(v -> query)).some(), is(10));
```

Comment:
Looks like we also need a static constructor taking a `Try1<Connection, A, SQLException>`to avoid users of DB needing to instantiate an anonymous class.  This can be done in another issue.